sdk/loadtestservice/Azure.Developer.LoadTesting/src/Models/Test.cs

```diff
@@ -0,0 +1,245 @@
+        /// <summary> Pass fail criteria for a test. </summary>
+        public PassFailCriteria PassFailCriteria
+        {
+            get
+            {
+                if (_passFailCriteria == null)
+                {
+                    _element.SetProperty("passFailCriteria", new { });
+                    _passFailCriteria = new PassFailCriteria(_element.GetProperty("passFailCriteria"));
+                }
+
+                return _passFailCriteria;
+            }
+
+            // TODO: need to test what happens if caller sets this, how the interaction
+            // across MJEs in the MJD works.
+            set { _passFailCriteria = value; }
+        }
+
+        private IDictionary<string, Secret> _secrets;
+        /// <summary> Secrets can be stored in an Azure Key Vault or any other secret store. If the secret is stored in an Azure Key Vault, the value should be the secret identifier and the type should be AKV_SECRET_URI. If the secret is stored elsewhere, the secret value should be provided directly and the type should be SECRET_VALUE. </summary>
+        public IDictionary<string, Secret> Secrets
+        {
+            get
+            {
+                if (_secrets == null)
+                {
+                    _element.SetProperty("secrets", new { });
+                    _secrets = new MutableJsonDictionary<Secret>(_element.GetProperty("secrets"));
+                }
+
+                return _secrets;
+            }
+        }
+
+        /// <summary> Certificates metadata. </summary>
+        public CertificateMetadata Certificate { get; set; }
+
+        private IDictionary<string, string> _environmentVariables;
+        /// <summary> Environment variables which are defined as a set of &lt;name,value&gt; pairs. </summary>
+        public IDictionary<string, string> EnvironmentVariables
+        {
+            get
+            {
+                if (_environmentVariables == null)
+                {
+                    _element.SetProperty("environmentVariables", new { });
+                    _environmentVariables = new MutableJsonDictionary<string>(_element.GetProperty("environmentVariables"));
+                }
+
+                return _environmentVariables;
+            }
+        }
+
+        /// <summary> The load test configuration. </summary>
+        public LoadTestConfiguration LoadTestConfiguration { get; set; }
+
+        /// <summary> The input artifacts for the test. </summary>
+        public TestInputArtifacts InputArtifacts { get; }
+
+        /// <summary> Unique test name as identifier. </summary>
+        public string TestId
+        {
+            get
+            {
+                if (_element.TryGetProperty("testId", out MutableJsonElement value))
+                {
+                    return value.GetString();
+                }
+                return null;
+            }
+            set => _element.SetProperty("testId", value);
+        }
```

Comment:
How could I tell if a property is actual `null` or I haven't set it? If I understand correctly,
```C#
var test = new Test();
test.TestId = null;
test.TestId == null ? true : false; // It is true
```
and 
```C#
var test = new Test();
test.TestId == null ? true : false; // It is true
```
That means, when we get a `Test`, before we send out it to service, we might want to check whether testId is set. But we cannot.

Reply:
Yes, from outside the model, you can't tell if a property has been changed.  Internally, MutableJsonElement holds a list of changes, and it uses that to write the PATCH JSON.  You can learn more about this in the [DynamicData blog post](https://devblogs.microsoft.com/azure-sdk/dynamic-json-in-azure-core/), where it talks about MutableJsonElement.

Reply:
You cannot expect users to check that from the change list of MutableJsonElement, right? I think we need a way to let users check `null`.